web/src/engine/websites/ProjectSuki.ts

```diff
@@ -0,0 +1,72 @@
+import { Tags } from '../Tags';
+import icon from './ProjectSuki.webp';
+import { DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise ((resolve, reject) => {
+        try {
+
+            const element = document.querySelector('.strip-reader');
+            const bookid = window.location.href.split('/') [4];
+            const chapterid = window.location.href.split('/') [5];
+            const body = JSON.stringify({
+                bookid: bookid,
+                chapterid: chapterid,
+                first: true
+            });
+
+            const xhr = new XMLHttpRequest();
+            xhr.onload = function () {
+                if (xhr.status >= 200 && xhr.status < 300) {
+                    const jsondata =  JSON.parse(xhr.response);
+
+                    const images = [];
+                    images.push(...[...document.querySelectorAll('img.img-fluid')].map(image => image.src));
```

Comment:
Can be simplified by assigning the entries directly instead of pushing into an empty array `const images = [...document/*...*/].map(/*...*/);`